rust/crd/src/lib.rs

```diff
@@ -87,45 +90,61 @@ pub enum Error {
 )]
 #[serde(rename_all = "camelCase")]
 pub struct NifiSpec {
-    /// The NiFi image to use
-    pub image: ProductImage,
-    /// Global Nifi config for e.g. authentication or sensitive properties
+    /// Settings that affect all roles and role groups.
+    /// The settings in the `clusterConfig` are cluster wide settings that do not need to be configurable at role or role group level.
     pub cluster_config: NifiClusterConfig,
-    /// Cluster operations like pause reconciliation or cluster stop.
-    #[serde(default)]
-    pub cluster_operation: ClusterOperation,
+
+    // no doc - docs in Role struct.
     #[serde(default, skip_serializing_if = "Option::is_none")]
-    /// Available NiFi roles
     pub nodes: Option<Role<NifiConfigFragment>>,
+
+    // no doc - docs in ProductImage struct.
+    pub image: ProductImage,
+
+    // no doc - docs in ClusterOperation struct.
+    #[serde(default)]
+    pub cluster_operation: ClusterOperation,
 }
 
 #[derive(Clone, Debug, Deserialize, JsonSchema, PartialEq, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct NifiClusterConfig {
-    /// Authentication options for NiFi (required)
+    /// Authentication options for NiFi (required).
+    /// Read more about authentication in the [security documentation](DOCS_BASE_URL_PLACEHOLDER/nifi/usage_guide/security).
     // We don't add `#[serde(default)]` here, as we require authentication
     pub authentication: Vec<NifiAuthenticationClassRef>,
-    /// Configuration options for how NiFi encrypts sensitive properties on disk
+
+    // no doc - docs in NifiSensitivePropertiesConfig struct.
     pub sensitive_properties: NifiSensitivePropertiesConfig,
-    /// Name of the Vector aggregator discovery ConfigMap.
+
+    /// Name of the Vector aggregator [discovery ConfigMap](DOCS_BASE_URL_PLACEHOLDER/concepts/service_discovery).
     /// It must contain the key `ADDRESS` with the address of the Vector aggregator.
+    /// Follow the [logging tutorial](DOCS_BASE_URL_PLACEHOLDER/tutorials/logging-vector-aggregator)
+    /// to learn how to configure log aggregation with Vector.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub vector_aggregator_config_map_name: Option<String>,
-    /// The reference to the ZooKeeper cluster
+
+    /// Kafka requires a ZooKeeper cluster connection to run.
```

Comment:
Becareful with the copy pasting :)
```suggestion
    /// NiFi requires a ZooKeeper cluster connection to run.
```